src/main/java/io/github/thebusybiscuit/hotbarpets/HotbarPetsListener.java

```diff
@@ -247,4 +234,25 @@ public void onDamage(EntityDamageEvent e) {
 		}
 	}
 
+	@EventHandler
+	public void onPhantomSpawn(EntityTargetLivingEntityEvent e) {
+		if (e.getEntityType() == EntityType.PHANTOM && ((Phantom) e.getEntity()).getTarget() instanceof Player
+				&& hasHotBarPet((Player) ((Phantom) e.getEntity()).getTarget(), panda)
+		) {
+			Player p = (Player) ((Phantom) e.getEntity()).getTarget();
+			if (!panda.tryToConsumeFood(p))
+				return;
+
+			e.getEntity().remove();
+			e.setCancelled(true);
+		}
+	}
+
+	private boolean hasHotBarPet(Player player, HotbarPet pet) {
+		for (int i = 0; i < 9; i++) {
+			if (SlimefunManager.isItemSimilar(player.getInventory().getItem(i), pet.getItem(), true))
```

Comment:
This could be optimized by using pet.isItem(...getItem(i)).
Thatd use the new id-system rather than comparing all itemmeta stuff. Because that is really slow.